draft-decraeneginsberg-lsr-isis-fast-flooding.xml

```diff
@@ -276,55 +274,57 @@
+			<t>Type: TBD1</t>
+			<t>Length: variable, the size in octets of the Value field</t>
+			<t>Value: a list of sub-TLVs</t>
 			<t>Two sub-TLVs are defined in this document. The support of any sub-TLV is OPTIONAL.</t>
-			<t>For a given IS-IS adjacency, the Flooding Parameters TLV does not need to be advertised in each IIH or PSNP. The IS transmitter uses the latest received value for each parameter (sub-TLV) until a new value is advertised by the IS receiver. Note however that IIH and PSNP are not reliably exchanged, hence may never be received. For a parameter which has never been advertised, the IS transmitter use its local default value. That value SHOULD be configurable on a per node basis and MAY be configurable on a per interface basis.</t>
-			<section anchor="LSPReceptionWindow" title="InterfaceLSPReceiveWindow sub-TLV">
-				<t>The sub-TLV InterfaceLSPReceiveWindow advertises the maximum number of un-acknowledged LSPs that the node can receive/process with no separation interval between LSPs. This is meant to be the size of the LSP input queue of the IS-IS process in the control plane.</t>
-				<t>Type: 1.</t>
-				<t>Length: 4 octets.</t>
-				<t>Value: number of un-acknowledged LSPs which can be sent back to back.</t>
-				<t>Note that if an LSP has not been acknowledged and is sent again, it does not count twice. The reason is that this LSP is assumed to be lost and hence not in a buffer at the receiver.</t>
+			<t>For a given IS-IS adjacency, the Flooding Parameters TLV does not need to be advertised in each IIH or PSNP. The LSP transmitter uses the latest received value for each parameter until a new value is advertised by the LSP receiver. However, IIHs and PSNPs are not reliably exchanged, and may never be received, so parameters MAY be sent even if there is no change in value since the last transmission. For a parameter which has never been advertised, the LSP transmitter SHOULD use its local default value. That value SHOULD be configurable on a per node basis and MAY be configurable on a per interface basis.</t>
+			<section anchor="LSPReceptionWindow" title="LSP Burst Window sub-TLV">
+				<t>The LSP Burst Window sub-TLV advertises the maximum number of un-acknowledged LSPs that the node can receive with no separation interval between LSPs.</t>
+				<t>Type: 1</t>
+				<t>Length: 4 octets</t>
+				<t>Value: number of un-acknowledged LSPs that can be sent back to back</t>
+				<t>Note that if an LSP has not been acknowledged and is sent again, it does not count twice. The reason is that this LSP is assumed to be lost and hence not in a buffer at the receiver.
+				[This requires some clarification. Why would we send the same LSP twice in the same burst? If it is a retransmission of an LSP that was originally sent outside of the current burst, then shouldn't it count as part of the burst? -- Tony L.]</t>
```

Comment:
I don't think this paragraph is needed.

Reply:
I think that this paragraph is needed. One use case:
- LSP Burst Window is N
- sender sends N LSPs which are all lost on the way. 
- after the time-out, the sender needs to be able to re-sent those LSP. Otherwise As the available window is empty forever.
Another way to present it:
- loss of LSP mechanically reduce the available size of the window as those LSP will never be acknowledged/received. At the timeout, we assume that those LSP are lost hence not the input buffer of the receiver.
- we could rephrase to say that 'LSP Burst Window' is the maximum number of LSP which are both unacknowledged and not time-out.

Reply:
It seems that you want to define a relationship between Burst and RWIN - but the text is not clear.
I'll let you and Tony hammer this one out since he also seems to not be completely satisfied w the current textx.

Reply:
Yes there is a relationship between RWIN and Burst. It is meant to be defined in the RWIN algo. Guillaume will propose a new text so I suggest that we wait for this new text.

Reply:
Well we should be careful with RWin, RWin is indeed equivalent to an allowed burst **if** there is no additional mechanism to control the sending -which might very well be the case-.
Maybe the following wording could work:
"A sender must take special care in counting the unacknowledged LSPs: when an LSP needs to be retransmitted because of the timeout, it is not counted in the RWin anymore until it is effectively retransmitted." since I think this is the pitfall we want to avoid ?

Reply:
I don't understand the relationship that you're trying to create between RWin and the burst size.

The burst size is simply what the receiver can handle. Period.

It seems like you're also trying to describe a behavior about how the transmitter behaves.  Fair enough, except that we need to be explicit and clear about that behavior.

Proposed text: "After sending a burst, the transmitter must maintain a count of the unacknowledged LSPs. When the transmitter receives a PSNP as an acknowledgment, the number of LSPs in the PSNP should be subtracted from this count. If no acknowledgment for an LSP is received within the LSP timeout, the count of unacknowledged LSPs should be decremented. After each change, the transmitter MAY send a partial burst, adding the number of LSPs in the burst to the count of the unacknowledged LSPs. The count of unacknowledged LSPs must never exceed the LSP Burst Window."

If the count of unacknowledged LSPs is what you mean by RWin, that's fine, but we must be explicit about a definition.

Reply:
Another thought: do we want to advertise an RWin independently of the burst window?

Reply:
I agree that this needs clarification : I think that Bruno was saying that with the vanilla RWin algorithm, RWin is the burst size (ie you can send up to RWin LSPs in one time). But this is imho a particular case and your last proposal explains the general case better.

Small nit: "After sending a burst" -> actually, at all times ?

Reply:
The drawback of using a burst size != RWin is that the period between two burst should be defined. But I can't think of a simple way of setting it.

Reply:
I agree that this need to be crystal clear. Your proposed text is fine for me. Does someone intend to use this value for a different usage than RWIN? If not, why not simply call it 'Receive Window'. This would remove the point raised by Guillaume (burst over which period?).
BTW, we also need to clarify the meaning of RWIN: simplest option seem to call RWIN  the value advertised by the receiver (hence static from the sender POV, so there is no such thing as initial value of RWIN) and to use CWIN to adjust the window on the sender size. I'm